### PR TITLE
Add NewTokenClient

### LINKED ISSUE #2636: Add NewTokenClient constructor
I often write one-off tools using this module and I keep needing to come back to the docs to copy/paste this snippet.

``` go
return github.NewClient(
	oauth2.NewClient(
		ctx, oauth2.StaticTokenSource(
			&oauth2.Token{
				AccessToken: token,
			},
		),
	),
)
```

It would be nice if there was a helper to do this.

``` go
client := github.NewTokenClient(ctx, token)
```

### PR DESCRIPTION
**Pull request by icholy**
Fixes #2636

### CODE REVIEW

File: github/github.go
Comment (gmlewis) [inline github/github.go:350]:
```suggestion
// NewTokenClient returns a new GitHub API client authenticated with the provided token.
```

File: github/github.go
Comment (gmlewis) [inline github/github.go:359]: How about collapsing these 7 lines?

```suggestion
		oauth2.NewClient(ctx, oauth2.StaticTokenSource(	&oauth2.Token{AccessToken: token})),
```

**Review by gmlewis — CHANGES_REQUESTED** [review]

**Comment by icholy** [comment]
I removed the context parameter since I think most users won't be needing it. Let me know if you disagree.

**Review by gmlewis — APPROVED** [review]
Thank you, @icholy !
LGTM.

Awaiting second LGTM+Approval from any other contributor to the repo.

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by icholy** [comment]
Should I squash?

**Comment by gmlewis** [comment]
> Should I squash?

No need to squash, as we always "Squash and Merge" all PRs in this repo.  Thanks, though.

**Comment by gmlewis** [comment]
Do you want to update an example before merging, @icholy ?

**Comment by gmlewis** [comment]
> LGTM!

Thank you, @valbeat !

**Comment by icholy** [comment]
I'll do a separate PR later.

**Pull request merged**
